pywb/indexer/cdxindexer.py

```diff
@@ -50,11 +50,16 @@ def write(self, entry, filename):
         if not entry.get('url') or not entry.get('urlkey'):
             return
 
-        if entry.record.rec_type == 'warcinfo':
+        if self._is_skipped(entry):
             return
 
         self.write_cdx_line(self.out, entry, filename)
 
+    def _is_skipped(self, entry):
+        if entry.record.rec_type == 'warcinfo':
+            return True
+        return entry.record.rec_type == 'metadata' and entry['mime'] == 'text/anvl'
```

Comment:
Maybe should move this type to `METADATA_NO_INDEX_TYPES = ('text/anvl')` and do `entry['mime'] in METADATA_NO_INDEX_TYPES` in case we need to add other types later?

More broadly, maybe it makes sense to filter out a `metadata` record if it comes after a request/response pair and has the same url? The index is already checking for the pair, so maybe could also check if it encounters metadata if it was the same url? That should filter most of the Heritrix metadata records, I think..

Reply:
Agree `METADATA_NO_INDEX_TYPES` class var is a good idea, updated PR to include this.

It might make sense to filter out `metadata` when it comes in that specific order tho Heritrix 3 creates a metadata record after info (containing outlinks).
That might be an issue tho if we run into this issue some more that approach will be the way to go.

Could also add an option to include metadata records when indexing where the default behavior is to exclude them.